clients/src/main/java/org/apache/kafka/common/metrics/KafkaMetric.java

```diff
@@ -78,6 +111,10 @@ public Measurable measurable() {
         }
     }
 
+    /**
+     * Set the metric config.
```

Comment:
This method normally is used to update quota and it should be used by server-side only. Could you add comments to remind `MetricsReporter` users that they should NOT call this method? ditto for `config()` method

Reply:
I understand that for the config setter, users should not use it because it's aimed for server-side caller. However, why are users not allowed to fetch the config?

Reply:
> However, why are users not allowed to fetch the config?

`MetricConfig` has the methods used by server too, so encouraging user to access `MetricConfig` could lead potential trouble I feel.